### PR TITLE
Patch requested changes pr6

### PR DESCRIPTION
**Pull request by olivierverwoerd**
changed to feedback from the pullrequest 6

### CODE REVIEW

File: code/headers/gas_sensors_c.hpp [hunk 1]
```diff
@@ -48,16 +48,14 @@ namespace r2d2::gas_detection {
...
                 if (!frame.request && !has_send) {
                     continue;
                 }
-
                 for (gas_sensor_interface_c<AmountOfGasses> *sensor : sensors) {
                     std::array<r2d2::gas_detection::gas_s, AmountOfGasses>
                         sensor_data = sensor->get();
                     for (r2d2::gas_detection::gas_s gas : sensor_data) {
-                        frame_gas.gas_id = gas.gas_id;
+                        frame_gas.gas_id = uint8_t(gas.gas_id);
```
Comment (YouriMulder) [inline code/headers/gas_sensors_c.hpp]: Here you cast the gas type to an uint8_t and in the operator<< for gas you cast it to an int. I think this is a bit inconsistent.

Comment (YouriMulder) [reply code/headers/gas_sensors_c.hpp]: This is a c style case. You should use c++ static_cast

File: code/headers/gas_s.hpp [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
+        gas_type gas_id;
         int value;
+
         /**
          * An operator which outputs the gas_id and the value in parts per
          * million.
          */
         friend hwlib::ostream &operator<<(hwlib::ostream &out, const gas_s &g) {
-            return out << "gas_id: " << g.gas_id << " value: " << g.value;
+            return out << "gas_id: " << int(g.gas_id) << " value: " << g.value;
```
Comment (YouriMulder) [inline code/headers/gas_s.hpp]: This is a c style cast. You should use static_cast.

**Review by YouriMulder — CHANGES_REQUESTED** [review]
I think the overall documentation is still pretty weak. I don't see any @param @return I think this might be an issue when creating a PR to master.

Your changes look fine. Please change the casts to static_casts. Using this instead of c style casts is recommended in the R2D2 project.

**Comment by olivierverwoerd** [comment]
The uint 16 is because the communucatie frame
The doxygen shouldnt be a problem since this is a patch brange

**Review by YouriMulder — APPROVED** [review]

**Comment by YouriMulder** [comment]
> 
> 
> The uint 16 is because the communucatie frame
> The doxygen shouldnt be a problem since this is a patch brange

Ok then I understand why it is like this.

**Pull request merged**